### ISSUE TITLE
Fail the travis CI if the code is not formatted 

### ISSUE THREAD

**Issue by kunall17** [zulip/zulip-android#427]
[Read our guide on bug reports here](./ISSUES_GUIDE.md)

**Actual Behaviour**

It would be great if somehow we can fail the travis CI check if the code is not formatted,
I googled a bit and found these two but for different languages 
https://groups.google.com/forum/#!topic/golang-nuts/pdrN4zleUio
https://github.com/sbt/sbt/issues/2657

There can be a possiblity that this is not possible with android project as the Android Studio (or rather intelliJ) does the code formatting, hence we can close this issue! 

**Expected Behaviour**

Explain what you expected (unnecessary if it's obviouxly broken, e.g. a crash).

**Would you like to work on the issue?**

Let us know if this issue should be assigned to you.

**Comment by divs4debu** [comment]
I was trying to solve this issue and in research for this i emailed travis developers. This is the screen shot of their reply
![photo6280457164736800687](https://cloud.githubusercontent.com/assets/9443348/24687382/f1ee56e0-19d6-11e7-9d7c-b014fd4637d4.jpg)

After researching for the solution i found this tool-
https://github.com/checkstyle/checkstyle

I am working on this. I would like to have your suggestions too so i could speed up my work.
@kunall17 @saketkumar95 @Sam1301 @timabbott

**Comment by divs4debu** [comment]
@kunall17 any review over this?

**Comment by divs4debu** [comment]
@kunall17 Please review.

**Comment by kunall17** [comment]
Awesome, this seems promising! 👍 
By any chance you did some testing with the library you mentioned? 

Apologies for the late reply!

**Comment by vishwesh3** [comment]
(There are some inbuilt code patterns checks in Codacy )
https://support.codacy.com/hc/en-us/articles/207994335-Code-Patterns

**Comment by divs4debu** [comment]
@kunall17  I was waiting for someone to approve my findings , Now i will start working on it.

**Comment by divs4debu** [comment]
@vishwesh3 Yup codacy provide that support . I think we can have discussion for integration of codacy too, but if we are able to do that with travis ci that would be better option .
